Index: vendor/Twig/lib/Twig/Loader.php
===================================================================
--- vendor/Twig/lib/Twig/Loader.php	(revision 79)
+++ vendor/Twig/lib/Twig/Loader.php	(working copy)
@@ -85,6 +85,13 @@
 
         return $cls;
       }
+      try {
+        $contents = $this->compile($template, $name);
+      } catch (Exception $e) {
+        fclose($fp);
+        unlink($cache);
+        throw $e;
+      }
       file_put_contents($cache, $this->compile($template, $name));
       fclose($fp);
     }
